src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -1295,6 +1295,21 @@ public void run()
             }
          }.schedule(100);
       }
+      
+      // restore cursor position (if available)
+      if (document.getProperties().hasKey("selection"))
+      {
+         final String rangeSpec = document.getProperties().getString("selection");
+         final Range range = Range.decode(rangeSpec);
+         new Timer()
+         {
+            @Override
+            public void run()
+            {
+               docDisplay_.setSelectionRange(range);
+            }
+         }.schedule(100);
```

Comment:
It looks like this is going to stomp on the source position used for popin/popout since it runs on a timer that will fire after we've already set the cursor position there. It probably shouldn't do that since the popin/popout source position is more accurate. Can we defer to that position if it's available, or unify the mechanisms so they do not conflict? 

https://github.com/rstudio/rstudio/blob/0d7b3def3844265a4d807351b3f3f17bc489fbeb/src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java#L843-L844

https://github.com/rstudio/rstudio/blob/0d7b3def3844265a4d807351b3f3f17bc489fbeb/src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java#L1883